### PR TITLE
Edit a review inline on the listing page

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-frontend/issues/5944

<details>
<summary>Screencast</summary>

addons-frontend-inline-edit mov

</details>

### CODE REVIEW

**Comment by willdurand** [comment]
Isn't it a bit weird to allow review editing after a reply to the review has been left?

**Comment by kumar303** [comment]
> Isn't it a bit weird to allow review editing after a reply to the review has been left?

Editing a review in general can be confusing, especially since developers get email notifications with the original text. 

We have this on file to address it: https://github.com/mozilla/addons-frontend/issues/2043 The decision was to just say *edited* next to the date. A fancier solution would be to do it like github where you can view all previous versions.

File: src/amo/components/AddonReviewManager/index.js [hunk 1]
```diff
@@ -113,6 +114,7 @@ export class AddonReviewManagerBase extends React.Component<InternalProps> {
         <DismissibleTextForm
           formFooter={formFooter}
           isSubmitting={flashMessage === STARTED_SAVE_REVIEW}
+          onDismiss={onCancel || undefined}
```
Comment (bobsilverberg) [inline src/amo/components/AddonReviewManager/index.js]: Won't `onCancel` be `undefined` if it's falsey?

File: tests/unit/amo/components/TestAddonReviewCard.js [hunk 2]
```diff
@@ -806,9 +809,60 @@ describe(__filename, () => {
...
+      store.dispatch(showEditReviewForm({ reviewId: review.id }));
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+
+      const root = renderInline({ review });
+
+      const manager = root.find(AddonReviewManager);
+      expect(manager).toHaveProp('onCancel');
+
+      dispatchSpy.resetHistory();
+      const onCancel = manager.prop('onCancel');
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestAddonReviewCard.js]: Nit: this `const` seems a bit redundant.

**Comment by kumar303** [comment]
> When editing an existing review, the button says "Submit review" but I think "Save review" might be better

Good idea. I added it.

> It feels a bit odd to be able to edit the rating by clicking on the stars with the box open ... I can see how it might be confusing to a user

I think the *Saved* notification will help address this so I'm not too concerned. Let's get some feedback on the UI as is and see what Phil thinks.

> do you think we should consider also allowing a person to edit their rating by clicking on the small stars even with the inline form not visible?

I guess we could? I'm not too worried about it because clicking *Edit review* is a pretty intuitive way to do it if you want to change your rating from the listing page.

File: tests/unit/amo/components/TestAddonReviewCard.js (hunk 2 above)
Comment (kumar303) [reply tests/unit/amo/components/TestAddonReviewCard.js]: I don't have a strong opinion but, to me, `manager.prop('onCancel')()` is harder to read than `onCancel()`.

**Pull request merged**